cmake/compiler.cmake

```diff
@@ -0,0 +1,41 @@
+set_property(GLOBAL PROPERTY CXX_STANDARD 17)
+set_property(GLOBAL PROPERTY CXX_STANDARD_REQUIRED ON)
+set_property(GLOBAL PROPERTY CXX_EXTENSIONS ON)
+set_property(GLOBAL PROPERTY LINKER_LANGUAGE CXX)
+
+add_definitions(-DFMT_HEADER_ONLY)
+
+if(UNIX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -Werror -WFatal-errors -fsigned-char")
```

Comment:
Consider add_compiler_options instead, -std=c++17 is redundant due to CXX_STANDARD above

Reply:
Also -WFatal-errors should be -Wfatal-errors

Reply:
I need to fix up some of the higher up variables to get the std fixed.